Services/UserService/UserService.cs

```diff
@@ -0,0 +1,141 @@
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Adds a new user.
+        /// </summary>
+        /// <param name="user">The user to add.</param>
+        /// <returns>A list of all users including the newly added user.</returns>
+        public async Task<List<UsersDto>> AddUser(UsersDto newUser)
+        {
+            Users user = _mapper.Map<Users>(newUser);
+
+            // Assign a unique ID to the user
+            if (users.Any())
+            {
+                user.Id = users.Max(u => u.Id) + 1;
+            }
+            else
+            {
+                user.Id = 1;
+            }
+
+            // Add the user to the in-memory storage
+            users.Add(user);
+
+            return (users.Select(u => _mapper.Map<UsersDto>(u))).ToList();
+        }
+
+        /// <summary>
+        /// Updates an existing user.
+        /// </summary>
+        /// <param name="id">The ID of the user to update.</param>
+        /// <param name="updatedUser">The updated user details.</param>
+        /// <returns>The updated user object, or null if the user was not found.</returns>
+        public async Task<UsersDto> UpdateUser(int id, UsersDto updatedUser)
+        {
+            try
+            {
+                // Find the user to update by ID
+                Users user = users.FirstOrDefault(u => u.Id == id);
+                if (user != null)
+                {
+                    // Update the user 
+                    user.Name = updatedUser.Name;
+                    user.Email = updatedUser.Email;
+                    user.Password = updatedUser.Password;
+                    return _mapper.Map<UsersDto>(user);
+                }
+                else
+                {
+                    throw new Exception("User not found.");
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Deletes a user by ID.
+        /// </summary>
+        /// <param name="id">The ID of the user to delete.</param>
+        /// <returns>The deleted user object, or null if the user was not found.</returns>
+        public async Task<List<UsersDto>> DeleteUser(int id)
+        {
+            try
+            {
+                // Find the user to delete by ID
+                Users user = users.FirstOrDefault(u => u.Id == id);
+                if (user != null)
+                {
+                    // Remove the user from the in-memory storage
+                    users.Remove(user);
+
+                    return (users.Select(c => _mapper.Map<UsersDto>(c))).ToList();
+                }
+                else
+                {
+                    throw new Exception("User not found.");
+                }
+            }
+            catch (Exception)
+            {
```

Comment:
remove try catch from service if you are only throwing it.

Reply:
done